README.md
# UVB-76-Decoder 1.0-rc8

## Requirements:

    pip install -r requirements.txt

## Compile:

    pyinstaller --name UVB-76-Decoder-v1.0.0-x86_64 --onefile --noconsole --windowed  main.py


CHANGELOG.md
----------

### v1.0.0:

- The UVB-76 Decoder is a software tool designed to capture and decode the mysterious UVB-76 radio transmissions. This radio, also known as "Buzzer", is famous for transmitting sequences of numbers and intriguing sounds, generating speculations and theories all over the world.

- Message decoding: The UVB-76 Decoder uses algorithms to decode radio messages. It employs cryptography, analysis and pattern recognition techniques to try to decipher the hidden meaning behind sequences of numbers.

- ----------


src/aboult.py
import base64
from tkinter import *
from tkinter import messagebox
from tkinter import ttk


def help_info():
   """displays information about the program.
   clicking on the button will open a window with information.
   """
    
   window = Tk()
   window.title("UVB-76 Decoder")
   window.geometry("435x300")
   window.resizable(False, False)
   window.attributes('-alpha',9.1)

   custom_font_name = ('Arial', 15)
   label = Label(window,
                text="UVB-76 Decoder",
                font=custom_font_name).place(x=150, y=20)
   
   custom_font_version = ('Arial', 14)
   label = Label(window,
                text="v1.0.0",
                font=custom_font_version).place(x=188, y=58)


   label = Label(window,
                text="This software comes with absolutely no warranty.", ).place(x=45, y=154)

   label = Label(window, 
                text="For more details, visit the GNU General Public License, version 2", ).place(x=9, y=175)
                
   label = Label(window,
                text="Copyright © 2023  Juan Bindez",).place(x=120, y=260)

src/decode_pt_module.py


message_text_pt = {
    1: "Bem-aventurado o homem que não anda segundo o conselho dos ímpios, nem se detém no caminho dos pecadores, nem se assenta na roda dos escarnecedores",
    2: "Por que se amotinam os gentios, e os povos imaginam coisas vàs?",
    3: "Senhor, como se têm multiplicado os meus adversários! São muitos os que se levantam contra mim.",
    4: "Ouve-me quando eu clamo, ó Deus da minha justiça, na angústia me deste largueza; tem misericórdia de mim e ouve a minha oração.",
    5: "DÁ ouvidos às minhas palavras, ó Senhor, atende à minha meditação.",
    6: "Senhor, não me repreendas na tua ira, nem me castigues no teu furor.",
    7: "Senhor meu Deus, em ti confio; salva-me de todos os que me perseguem, e livra-me;",
    8: "Ó Senhor, Senhor nosso, quão admirável é o teu nome em toda a terra, pois puseste a tua glória sobre os céus!",
    9: "Eu te louvarei, Senhor, com todo o meu coração; contarei todas as tuas maravilhas.",
    10: "Por que estás ao longe, SENHOR? Por que te escondes nos tempos de angústia?",
    11: "No SENHOR confio; como dizeis à minha alma: Fugi para a vossa montanha como pássaro?",
    12: "Salva-nos, SENHOR, porque faltam os homens bons; porque são poucos os fiéis entre os filhos dos homens.",
    13: "Até quando te esquecerás de mim, SENHOR? Para sempre? Até quando esconderás de mim o teu rosto?",
    14: "Disse o néscio no seu coração: Não há Deus. Têm-se corrompido, fazem-se abomináveis em suas obras, não há ninguém que faça o bem.",
    15: "SENHOR, quem habitará no teu tabernáculo? Quem morará no teu santo monte?",
    16: "Guarda-me, ó Deus, porque em ti confio.",
    17: "Ouve, SENHOR, a justiça; atende ao meu clamor; dá ouvidos à minha oração, que não é feita com lábios enganosos.",
    18: "Eu te amarei, ó SENHOR, fortaleza minha.",
    19: "Os céus declaram a glória de Deus e o firmamento anuncia a obra das suas mãos.",
    20: "O SENHOR te ouça no dia da angústia, o nome do Deus de Jacó te proteja.",
    21: "O rei se alegra em tua força, SENHOR; e na tua salvação grandemente se regozija.",
    22: "Deus meu, Deus meu, por que me desamparaste? Por que te alongas do meu auxílio e das palavras do meu bramido?",
    23: "O SENHOR é o meu pastor, nada me faltará.",
    24: "Do SENHOR é a terra e a sua plenitude, o mundo e aqueles que nele habitam.",
    25: "A ti, SENHOR, levanto a minha alma.",
    26: "Julga-me, SENHOR, pois tenho andado em minha sinceridade; tenho confiado também no SENHOR; não vacilarei.",
    27: "O SENHOR é a minha luz e a minha salvação; a quem temerei? O SENHOR é a força da minha vida; de quem me recearei?",
    28: "A ti clamarei, ó Senhor, Rocha minha; não emudeças para comigo; não aconteça, cal ando-te tu para comigo, que eu fique semelhante aos que descem ao abismo.",
    29: "Dai ao SENHOR, ó filhos dos poderosos, dai ao SENHOR glória e força.",
    30: "Exaltar-te-ei, ó SENHOR, porque tu me exaltaste; e não fizeste com que meus inimigos se alegrassem sobre mim.",
    31: "Em ti, SENHOR, confio; nunca me deixes confundido. Livra-me pela tua justiça.",
    32: "Bem-aventurado aquele cuja transgressão é perdoada, e cujo pecado é coberto.",
    33: "Regozijai-vos no SENHOR, vós justos, pois aos retos convém o louvor.",
    34: "Louvarei ao SENHOR em todo o tempo; o seu louvor estará continuamente na minha boca.",
    35: "Pleiteia, SENHOR, com aqueles que pleiteiam comigo; peleja contra os que pelejam contra mim.",
    36: "A transgressão do ímpio diz no íntimo do meu coração: Não há temor de Deus perante os seus olhos.",
    37: "NÃO te indignes por causa dos malfeitores, nem tenhas inveja dos que praticam a iniqüidade.",
    38: "Ó Senhor, não me repreendas na tua ira, nem me castigues no teu furor.",
    39: "Eu disse: Guardarei os meus caminhos para não pecar com a minha língua; guardarei a boca com um freio, enquanto o ímpio estiver diante de mim.",
    40: "Esperei com paciência no SENHOR, e ele se inclinou para mim, e ouviu o meu clamor.",
    41: "Bem-aventurado é aquele que atende ao pobre; o SENHOR o livrará no dia do mal.",
    42: "Assim como o cervo brama pelas correntes das águas, assim suspira a minha alma por ti, ó Deus!",
    43: "Faze-me justiça, ó Deus, e pleiteia a minha causa contra a nação ímpia. Livra-me do homem fraudulento e injusto.",
    44: "Ó Deus, nós ouvimos com os nossos ouvidos, e nossos pais nos têm contado a obra que fizeste em seus dias, nos tempos da antiguidade.",
    45: "O meu coração ferve com palavras boas, falo do que tenho feito no tocante ao Rei. A minha língua é a pena de um destro escritor.",
    46: "Deus é o nosso refúgio e fortaleza, socorro bem presente na angústia.",
    47: "Batei palmas, todos os povos; aclamai a Deus com voz de triunfo.",
    48: "Grande é o SENHOR e mui digno de louvor, na cidade do nosso Deus, no seu monte santo.",
    49: "Ouvi isto, vós todos os povos; inclinai os ouvidos, todos os moradores do mundo,",
    50: "O Deus poderoso, o SENHOR, falou e chamou a terra desde o nascimento do sol até ao seu ocaso.",
    51: "Tem misericórdia de mim, ó Deus, segundo a tua benignidade; apaga as minhas transgressões, segundo a multidão das tuas misericórdias.",
    52: "Por que te glorias na malícia, ó homem poderoso? Pois a bondade de Deus permanece continuamente.",
    53: "Disse o néscio no seu coração: Não há Deus. Têm-se corrompido, e cometido abominável iniqüidade; não há ninguém que faça o bem.",
    54: "Salva-me, ó Deus, pelo teu nome, e faze-me justiça pelo teu poder.",
    55: "Inclina, ó Deus, os teus ouvidos à minha oração, e não te escondas da minha súplica.",
    56: "Tem misericórdia de mim, ó Deus, porque o homem procura devorar-me; pelejando todo dia, me oprime.",
    57: "Tem misericórdia de mim, ó Deus, tem misericórdia de mim, porque a minha alma confia em ti; e à sombra das tuas asas me abrigo, até que passem as calamidades.",
    58: "Acaso falais vós, deveras, ó congregação, a justiça? Julgais retamente, ó filhos dos homens?",
    59: "Livra-me, meu Deus, dos meus inimigos, defende-me daqueles que se levantam contra mim.",
    60: "Ó Deus, tu nos rejeitaste, tu nos espalhaste, tu te indignaste; oh, volta-te para nós.",
    61: "Ouve, ó Deus, o meu clamor; atende à minha oração.",
    62: "A minha alma espera somente em Deus; dele vem a minha salvação.",
    63: "Ó Deus, tu és o meu Deus, de madrugada te buscarei; a minha alma tem sede de ti; a minha carne te deseja muito em uma terra seca e cansada, onde não há água;",
    64: "Ouve, ó Deus, a minha voz na minha oração; guarda a minha vida do temor do inimigo.",
    65: "A ti, ó Deus, espera o louvor em Sião, e a ti se pagará o voto.",
    66: "Celebrai com júbilo a Deus, todas as terras.",
    67: "Deus tenha misericórdia de nós e nos abençoe; e faça resplandecer o seu rosto sobre nós (Selá.)",
    68: "Levante-se Deus, e sejam dissipados os seus inimigos; fugirão de diante dele os que o odeiam.",
    69: "Livra-me, ó Deus, pois as águas entraram até à minha alma.",
    691: "Livra-me, ó Deus, pois as águas entraram até à minha alma.",
    70: "Apressa-te, ó Deus, em me livrar; SENHOR, apressa-te em ajudar-me.",
    71: "Em ti, SENHOR, confio; nunca seja eu confundido.",
    72: "Ó Deus, dá ao rei os teus juízos, e a tua justiça ao filho do rei.",
    73: "Verdadeiramente bom é Deus para com Israel, para com os limpos de coração.",
    74: "Ó Deus, por que nos rejeitaste para sempre? Por que se acende a tua ira contra as ovelhas do teu pasto?",
    75: "A ti, ó Deus, glorificamos, a ti damos louvor, pois o teu nome está perto, as tuas maravilhas o declaram.",
    76: "Conhecido é Deus em Judá; grande é o seu nome em Israel.",
    77: "Clamei a Deus com a minha voz, a Deus levantei a minha voz, e ele inclinou para mim os ouvidos.",
    78: "Escutai a minha lei, povo meu; inclinai os vossos ouvidos às palavras da minha boca.",
    79: "Ó Deus, os gentios vieram à tua herança; contaminaram o teu santo templo; reduziram Jerusalém a montões de pedras.",
    80: "Tu, que és pastor de Israel, dá ouvidos; tu, que guias a José como a um rebanho; tu, que te assentas entre os querubins, resplandece.",
    81: "Exultai a Deus, nossa fortaleza; jubilai ao Deus de Jacó.",
    82: "Deus está na congregação dos poderosos; julga no meio dos deuses.",
    83: "Ó Deus, não estejas em silêncio; não te cales, nem te aquietes, ó Deus,",
    84: "Quão amáveis são os teus tabernáculos, SENHOR dos Exércitos!",
    85: "Abençoaste, SENHOR, a tua terra; fizeste voltar o cativeiro de Jacó.",
    86: "Inclina, SENHOR, os teus ouvidos, e ouve-me, porque estou necessitado e aflito.",
    87: "O seu fundamento está nos montes santos.",
    88: "SENHOR Deus da minha salvação, diante de ti tenho clamado de dia e de noite.",
    89: "As benignidades do SENHOR cantarei perpetuamente; com a minha boca manifestarei a tua fidelidade de geração em geração.",
    90: "SENHOR, tu tens sido o nosso refúgio, de geração em geração.",
    91: "Aquele que habita no esconderijo do Altíssimo, à sombra do Onipotente descansará.",
    92: "Bom é louvar ao SENHOR, e cantar louvores ao teu nome, ó Altíssimo;",
    93: "O SENHOR reina; está vestido de majestade. O SENHOR se revestiu e cingiu de poder; o mundo também está firmado, e não poderá vacilar.",
    94: "Ó Senhor Deus, a quem a vingança pertence, ó Deus, a quem a vingança pertence, mostra-te resplandecente.",
    95: "Vinde, cantemos ao SENHOR; jubilemos à rocha da nossa salvação.",
    96: "Cantai ao SENHOR um cântico novo, cantai ao SENHOR toda a terra.",
    97: "O SENHOR reina; regozije-se a terra; alegrem-se as muitas ilhas.",
    98: "Cantai ao SENHOR um cântico novo, porque fez maravilhas; a sua destra e o seu braço santo lhe alcançaram a salvação.",
    99: "O SENHOR reina; tremam os povos. Ele está assentado entre os querubins; comova-se a terra.",
    100: "Celebrai com júbilo ao SENHOR, todas as terras.",
    101: "Cantarei a misericórdia e o juízo; a ti, SENHOR, cantarei.",
    103: "Bendize, ó minha alma, ao SENHOR, e tudo o que há em mim bendiga o seu santo nome.",
    104: "Bendize, ó minha alma, ao SENHOR! SENHOR Deus meu, tu és magnificentíssimo; estás vestido de glória e de majestade.",
    105: "Louvai ao SENHOR, e invocai o seu nome; fazei conhecidas as suas obras entre os povos.",
    106: "Louvai ao SENHOR. Louvai ao SENHOR, porque ele é bom, porque a sua misericórdia dura para sempre.",
    107: "Louvai ao SENHOR, porque ele é bom, porque a sua benignidade dura para sempre.",
    108: "Preparado está o meu coração, ó Deus; cantarei e darei louvores até com a minha glória.",
    109: "Ó Deus do meu louvor, não te cales,",
    110: "Disse o SENHOR ao meu Senhor: Assenta-te à minha mão direita, até que ponha os teus inimigos por escabelo dos teus pés.",
    111: "Louvai ao SENHOR. Louvarei ao SENHOR de todo o meu coração, na assembléia dos justos e na congregação.",
    112: "Louvai ao SENHOR. Bem-aventurado o homem que teme ao SENHOR, que em seus mandamentos tem grande prazer.",
    113: "Louvai ao SENHOR. Louvai, servos do SENHOR, louvai o nome do SENHOR.",
    114: "Quando Israel saiu do Egito, e a casa de Jacó de um povo de língua estranha,",
    115: "Não a nós, SENHOR, não a nós, mas ao teu nome dá glória, por amor da tua benignidade e da tua verdade.",
    116: "Amo ao SENHOR, porque ele ouviu a minha voz e a minha súplica.",
    117: "Louvai ao SENHOR todas as nações, louvai-o todos os povos.",
    118: "Louvai ao SENHOR, porque ele é bom, porque a sua benignidade dura para sempre.",
    119: "Bem-aventurados os retos em seus caminhos, que andam na lei do Senhor.",
    120: "Na minha angústia clamei ao SENHOR, e me ouviu.",
    121: "Levantarei os meus olhos para os montes, de onde vem o meu socorro.",
    122: "Alegrei-me quando me disseram: Vamos à casa do Senhor.",
    123: "A ti levanto os meus olhos, ó tu que habitas nos céus.",
    124: "Se não fora o Senhor, que esteve ao nosso lado, ora diga Israel;",
    125: "Os que confiam no SENHOR serão como o monte de Sião, que não se abala, mas permanece para sempre.",
    126: "Quando o SENHOR trouxe do cativeiro os que voltaram a Sião, estávamos como os que sonham.",
    127: "Se o SENHOR não edificar a casa, em vão trabalham os que a edificam; se o SENHOR não guardar a cidade, em vão vigia a sentinela.",
    128: "Bem-aventurado aquele que teme ao SENHOR e anda nos seus caminhos.",
    129: "Muitas vezes me angustiaram desde a minha mocidade, diga agora Israel;",
    130: "Das profundezas a ti clamo, ó SENHOR.",
    131: "SENHOR, o meu coração não se elevou nem os meus olhos se levantaram; não me exercito em grandes matérias, nem em coisas muito elevadas para mim.",
    132: "Lembra-te, SENHOR, de Davi, e de todas as suas aflições.",
    133: "Oh! quão bom e quão suave é que os irmãos vivam em união.",
    134: "Eis aqui, bendizei ao SENHOR todos vós, servos do SENHOR, que assistis na casa do SENHOR todas as noites.",
    135: "Louvai ao SENHOR. Louvai o nome do SENHOR; louvai-o, servos do SENHOR.",
    136: "Louvai ao SENHOR, porque ele é bom; porque a sua benignidade dura para sempre.",
    137: "Junto aos rios da Babilônia, ali nos assentamos e choramos, quando nos lembramos de Sião.",
    138: "Eu te louvarei, de todo o meu coração; na presença dos deuses a ti cantarei louvores.",
    139: "SENHOR, tu me sondaste, e me conheces",
    140: "Livra-me, ó SENHOR, do homem mau; guarda-me do homem violento,",
    141: "SENHOR, a ti clamo, escuta-me; inclina os teus ouvidos à minha voz, quando a ti clamar.",
    142: "Com a minha voz clamei ao SENHOR; com a minha voz supliquei ao SENHOR.",
    143: "Ó Senhor, ouve a minha oração, inclina os ouvidos às minhas súplicas; escuta-me segundo a tua verdade, e segundo a tua justiça.",
    144: "Bendito seja o SENHOR, minha rocha, que ensina as minhas mãos para a peleja e os meus dedos para a guerra;",
    145: "Eu te exaltarei, ó Deus, rei meu, e bendirei o teu nome pelos séculos dos séculos e para sempre.",
    146: "Louvai ao Senhor. Ó minha alma, louva ao Senhor.",
    147: "Louvai ao SENHOR, porque é bom cantar louvores ao nosso Deus, porque é agradável; decoroso é o louvor.",
    148: "Louvai ao SENHOR. Louvai ao SENHOR desde os céus, louvai-o nas alturas.",
    149: "Louvai ao SENHOR. Cantai ao SENHOR um cântico novo, e o seu louvor na congregação dos santos.",
    150: "Louvai ao SENHOR. Louvai a Deus no seu santuário; louvai-o no firmamento do seu poder.",
}


src/decode_us_module.py



message_text_en = {
    1: "Blessed is the man who does not walk in the counsel of the wicked, nor stand in the path of sinners, nor sit in the seat of scoffers.",
    2: "Why do the nations rage, and the peoples plot in vain?",
    3: "O Lord, how many are my foes! Many are rising against me.",
    4: "Answer me when I call, O God of my righteousness! You have given me relief when I was in distress. Be gracious to me and hear my prayer.",
    5: "Give ear to my words, O Lord; consider my groaning.",
    6: "O Lord, rebuke me not in your anger, nor discipline me in your wrath.",
    7: "O Lord my God, in you I take refuge; save me from all my pursuers and deliver me,",
    8: "O Lord, our Lord, how majestic is your name in all the earth! You have set your glory above the heavens.",
    9: "I will give thanks to you, O Lord, with my whole heart; I will recount all of your wonderful deeds.",
    10: "Why, O Lord, do you stand far away? Why do you hide yourself in times of trouble?",
    11: "In the Lord I take refuge; how can you say to my soul, 'Flee like a bird to your mountain'?",
    12: "Save, O Lord, for the godly one is gone; for the faithful have vanished from among the children of man.",
    13: "How long, O Lord? Will you forget me forever? How long will you hide your face from me?",
    14: "The fool says in his heart, 'There is no God.' They are corrupt, doing abominable iniquity; there is none who does good.",
    15: "O Lord, who shall sojourn in your tent? Who shall dwell on your holy hill?",
    16: "Preserve me, O God, for in you I take refuge.",
    17: "O Lord, hear my righteousness; attend to my cry! Give ear to my prayer from lips free of deceit.",
    18: "I will love you, O Lord, my strength.",
    19: "The heavens declare the glory of God, and the sky above proclaims his handiwork.",
    20: "May the Lord answer you in the day of trouble! May the name of the God of Jacob protect you!",
    21: "The king rejoices in your strength, O Lord, and in your salvation how greatly he exults!",
    22: "My God, my God, why have you forsaken me? Why are you so far from saving me, from the words of my groaning?",
    23: "The Lord is my shepherd; I shall not want.",
    24: "The earth is the Lord's and the fullness thereof, the world and those who dwell therein.",
    25: "To you, O Lord, I lift up my soul.",
    26: "Judge me, O Lord, for I have walked in my integrity, and I have trusted in the Lord without wavering.",
    27: "The Lord is my light and my salvation; whom shall I fear? The Lord is the stronghold of my life; of whom shall I be afraid?",
    28: "To you, O Lord, I call; my rock, be not deaf to me, lest, if you be silent to me, I become like those who go down to the pit.",
    29: "Ascribe to the Lord, O heavenly beings, ascribe to the Lord glory and strength.",
    30: "I will extol you, O Lord, for you have drawn me up and have not let my foes rejoice over me.",
    31: "In you, O Lord, do I take.",
    31: "In you, O LORD, I trust; never let me be put to shame. Deliver me in your righteousness.",
    32: "Blessed is the one whose transgression is forgiven, whose sin is covered.",
    33: "Rejoice in the LORD, you righteous ones; praise befits the upright.",
    34: "I will bless the LORD at all times; his praise shall continually be in my mouth.",
    35: "Contend, O LORD, with those who contend with me; fight against those who fight against me.",
    36: "The wicked man's transgression speaks to him within his heart; there is no fear of God before his eyes.",
    37: "Do not fret because of evildoers or be envious of those who do wrong.",
    38: "O LORD, do not rebuke me in your anger or discipline me in your wrath.",
    39: "I said, 'I will guard my ways that I may not sin with my tongue; I will keep a muzzle on my mouth as long as the wicked are in my presence.'",
    40: "I waited patiently for the LORD; he inclined to me and heard my cry.",
    41: "Blessed is the one who has regard for the weak; in the day of trouble, the LORD delivers him.",
    42: "As the deer pants for streams of water, so my soul longs after you, O God.",
    43: "Vindicate me, O God, and plead my cause against an ungodly nation; deliver me from deceitful and unjust men.",
    44: "We have heard with our ears, O God; our fathers have told us what you did in their days, in days of old.",
    45: "My heart is stirred by a noble theme as I recite my verses for the king; my tongue is the pen of a skillful writer.",
    46: "God is our refuge and strength, an ever-present help in trouble.",
    47: "Clap your hands, all you peoples; shout to God with a voice of joy.",
    48: "Great is the LORD, and greatly to be praised, in the city of our God, His holy mountain.",
    49: "Hear this, all you peoples; listen, all inhabitants of the world,",
    50: "The Mighty One, God, the LORD, speaks and summons the earth from the rising of the sun to its setting.",
    51: "Have mercy on me, O God, according to your steadfast love; blot out my transgressions according to your abundant mercy.",
    52: "Why do you boast of evil, O mighty man? The steadfast love of God endures all day long.",
    53: "The fool says in his heart, 'There is no God.' They are corrupt, and their ways are vile; there is no one who does good.",
    54: "Save me, O God, by your name; vindicate me by your might.",
    55: "Incline your ear to my prayer, O God, and hide not yourself from my plea for mercy!",
    56: "Be gracious to me, O God, for man tramples on me; all day long an attacker oppresses me.",
    57: "Be merciful to me, O God, be merciful to me, for in you my soul takes refuge; in the shadow of your wings I will take shelter, till the storms of destruction pass by.",
    58: "Do you indeed decree what is right, you gods? Do you judge the children of man uprightly?",
    59: "Deliver me, O my God, from my enemies; protect me from those who rise up against me.",
    60: "O God, you have rejected us; you have broken us; you have been angry; oh, restore",
    61: "Hear, O God, my cry; attend to my prayer.",
    62: "My soul waits in silence for God alone; from Him comes my salvation.",
    63: "O God, You are my God; earnestly I seek You; my soul thirsts for You; my flesh faints for You, as in a dry and weary land where there is no water.",
    64: "Hear, O God, my voice in my prayer; preserve my life from fear of the enemy.",
    65: "To You, O God, belongs the praise in Zion; to You shall vows be performed.",
    66: "Shout for joy to God, all the earth.",
    67: "May God be gracious to us and bless us; may His face shine upon us (Selah).",
    68: "Arise, O God, and scatter your enemies; let those who hate You flee before You.",
    69: "Deliver me, O God, for the waters have come up to my neck.",
    691: "Deliver me, O God, for the waters have come up to my neck.",
    70: "Make haste, O God, to deliver me; O LORD, make haste to help me.",
    71: "In You, O LORD, I put my trust; let me never be put to shame.",
    72: "O God, give the king Your justice, and Your righteousness to the royal son.",
    73: "Truly God is good to Israel, to those who are pure in heart.",
    74: "O God, why have You rejected us forever? Why does Your anger smoke against the sheep of Your pasture?",
    75: "To You, O God, we give thanks; to You we give thanks, for Your name is near. Your wondrous deeds declare it.",
    76: "God is known in Judah; His name is great in Israel.",
    77: "I cry aloud to God, aloud to God, and He will hear me.",
    78: "Hear my teaching, O my people; incline your ears to the words of my mouth.",
    79: "O God, the nations have come into Your inheritance; they have defiled Your holy temple; they have laid Jerusalem in ruins.",
    80: "Give ear, O Shepherd of Israel, You who lead Joseph like a flock! You who are enthroned upon the cherubim, shine forth.",
    81: "Sing aloud to God our strength; shout for joy to the God of Jacob.",
    82: "God has taken His place in the divine council; in the midst of the gods, He holds judgment.",
    83: "O God, do not keep silence; do not hold Your peace or be still, O God!",
    84: "How lovely is Your dwelling place, O LORD of hosts!",
    85: "You have blessed, O LORD, Your land; You have restored the fortunes of Jacob.",
    86: "Incline Your ear, O LORD, and answer me, for I am poor and needy.",
    87: "His foundation is on the holy mountains.",
    88: "O LORD, God of my salvation, I cry out day and night before You.",
    89: "I will sing of the steadfast love of the LORD forever; with my mouth I will make known Your faithfulness to all generations.",
    90: "LORD, You have been our dwelling place in all generations.",
    91: "He who dwells in the shelter of the Most High will abide in the shadow of the Almighty.",
    92: "It is good to give thanks to the LORD, to sing praises to Your name, O Most High;",
    93: "The LORD reigns; He is clothed with majesty. The LORD has clothed Himself with strength; He has girded Himself. The world also is established, so that it cannot be moved.",
    94: "O Lord God, to whom vengeance belongs—O God, to whom vengeance belongs, shine forth!",
    95: "Oh come, let us sing to the LORD! Let us shout joyfully to the Rock of our salvation.",
    96: "Sing to the LORD a new song; sing to the LORD, all the earth.",
    97: "The LORD reigns; let the earth rejoice; let the multitude of isles be glad!",
    98: "Sing to the LORD a new song, for He has done marvelous things; His right hand and His holy arm have gained Him the victory.",
    99: "The LORD reigns; let the peoples tremble! He dwells between the cherubim; let the earth be moved!",
    100: "Make a joyful shout to the LORD, all you lands!",
    101: "I will sing of mercy and justice; to You, O LORD, I will sing praises.",
    103: "Bless the LORD, O my soul; and all that is within me, bless His holy name!",
    104: "Bless the LORD, O my soul! O LORD my God, You are very great: You are clothed with honor and majesty.",
    105: "Oh, give thanks to the LORD! Call upon His name; make known His deeds among the peoples!",
    106: "Praise the LORD! Oh, give thanks to the LORD, for He is good! For His mercy endures forever.",
    107: "Oh, give thanks to the LORD, for He is good! For His mercy endures forever.",
    108: "My heart is steadfast, O God; I will sing and give praise, even with my glory.",
    109: "Do not keep silent, O God of my praise!",
    110: "The LORD said to my Lord, 'Sit at My right hand, till I make Your enemies Your footstool.'",
    111: "Praise the LORD! I will praise the LORD with my whole heart, in the assembly of the upright and in the congregation.",
    112: "Praise the LORD! Blessed is the man who fears the LORD, who delights greatly in His commandments.",
    113: "Praise the LORD! Praise, O servants of the LORD, praise the name of the LORD!",
    114: "When Israel went out of Egypt, the house of Jacob from a people of strange language,",
    115: "Not unto us, O LORD, not unto us, but to Your name give glory, because of Your mercy, because of Your truth.",
    116: "I love the LORD, because He has heard my voice and my supplications.",
    117: "Praise the LORD, all you Gentiles! Laud Him, all you peoples!",
    118: "Oh, give thanks to the LORD, for He is good! For His mercy endures forever.",
    119: "Blessed are the undefiled in the way, who walk in the law of the LORD!",
    120: "In my distress I cried to the LORD, and He heard me.",
    121: "I will lift up my eyes to the hills—From whence comes my help?",
    122: "I was glad when they said to me, 'Let us go into the house of the LORD.'",
    123: "Unto You I lift up my eyes, O You who dwell in the heavens.",
    124: "If it had not been the LORD who was on our side—Let Israel now say—",
    125: "Those who trust in the LORD are like Mount Zion, which cannot be moved, but abides",
    126: "When the LORD brought back the captivity of Zion, we were like those who dream.",
    127: "Unless the LORD builds the house, they labor in vain who build it; unless the LORD guards the city, the watchman stays awake in vain.",
    128: "Blessed is everyone who fears the LORD, who walks in His ways.",
    129: "Many times they have afflicted me from my youth; yet they have not prevailed against me.",
    130: "Out of the depths I have cried to You, O LORD.",
    131: "LORD, my heart is not haughty, nor my eyes lofty. Neither do I concern myself with great matters, nor with things too profound for me.",
    132: "Remember, LORD, the reproach of Your servant David, how I bear in my bosom the reproach of all the many peoples.",
    133: "Behold, how good and how pleasant it is for brethren to dwell together in unity!",
    134: "Behold, bless the LORD, all you servants of the LORD, who by night stand in the house of the LORD!",
    135: "Praise the LORD! Praise the name of the LORD; praise Him, O you servants of the LORD!",
    136: "Praise the LORD, for He is good; for His mercy endures forever.",
    137: "By the rivers of Babylon, there we sat down, yea, we wept when we remembered Zion.",
    138: "I will praise You with my whole heart; before the gods I will sing praises to You.",
    139: "O LORD, You have searched me and known me.",
    140: "Deliver me, O LORD, from evil men; preserve me from violent men,",
    141: "LORD, I cry out to You; make haste to me! Give ear to my voice when I cry out to You.",
    142: "With my voice I cry out to the LORD; with my voice I plead for mercy to the LORD.",
    143: "Hear my prayer, O LORD; give ear to my supplications! In Your faithfulness and righteousness, answer me.",
    144: "Blessed be the LORD my Rock, who trains my hands for war, and my fingers for battle.",
    145: "I will extol You, my God, O King; and I will bless Your name forever and ever.",
    146: "Praise the LORD! O my soul, praise the LORD!",
    147: "Praise the LORD, for it is good to sing praises to our God; for it is pleasant, and praise is beautiful.",
    148: "Praise the LORD! Praise the LORD from the heavens; praise Him in the heights!",
    149: "Praise the LORD! Sing to the LORD a new song, and His praise in the assembly of saints.",
    150: "Praise the LORD! Praise God in His sanctuary; praise Him in His mighty firmament.",
}


main.py
# this is part of the UVB-76-Decoder project.
#
# Release: v1.1-rc1
#
# Copyright © 2023  Juan Bindez  <juanbindez780@gmail.com>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


import os
import time
import datetime

from tkinter import *
from tkinter import ttk
from tkinter import messagebox
from tkinter import font
from tkinter import messagebox

from src.decode_pt_module import *
from src.decode_us_module import *
from src.check_update_module import *
from src.aboult import *


window = Tk()
window.title("UVB-76 Decoder")
window.geometry("500x370")
#window['background'] = '#373636'  
window.resizable(False, False)# False for non-responsive window and True for responsive.
window.attributes('-alpha',9.1)
#foto_icon = PhotoImage(data=base64.b64decode(ICON_LOGO))
#window.iconphoto(True, foto_icon)


def decode_message():

    code = entrada_de_dados.get()
    numeros = [int(num) for num in code.split(" ")]
    name_file = str(datetime.datetime.now())
    arquivo_saida = open(f"UVB-76-code-({numeros}).doc", "w")

    if var_portugues.get() == 1:
        for num in numeros:
            if num in message_text_pt:
                texto = message_text_pt[num]
                arquivo_saida.write(texto+"\n")

                print(message_text_pt[num])
            else:
                messagebox.showerror("UVB-76 Decoder", "Número não possui um texto correspondente.")

    elif var_ingles.get() == 1:
        for num in numeros:
            if num in message_text_en:
                texto = message_text_en[num]
                arquivo_saida.write(texto+"\n")

                print(message_text_en[num])
            else:
                print("Número não possui um texto correspondente.")

    elif var_portugues.get() == 0 and var_ingles.get() == 0:
        messagebox.showerror("UVB-76 Decoder", "Check a language!")

    messagebox.showinfo("UVB-76 Decoder",
                                    "A .txt file was generated, there will be the decoding")


var_portugues = IntVar()
var_ingles = IntVar()

check_portugues = Checkbutton(window,
                         text="Portuguese",
                         bd=0,
                         variable=var_portugues,).place(x=80, y=220)

check_ingles = Checkbutton(window,
                        text="English",
                        bd=0,
                        variable=var_ingles).place(x=160, y=220)


def make_menu(w):
    global the_menu_1
    the_menu_1 = Menu(w, tearoff=0)
    the_menu_1.add_command(label="Colar")
    
    
def show_menu(e):
    w = e.widget
    the_menu_1.entryconfigure("Colar",
    command=lambda: w.event_generate("<<Paste>>"))
    the_menu_1.tk.call("tk_popup", the_menu_1, e.x_root, e.y_root)


make_menu(window)
entrada_de_dados = Entry(window, width=40)
entrada_de_dados.place(x=95, y=170)
entrada_de_dados.bind_class("Entry", "<Button-3><ButtonRelease-3>", show_menu)
lbl = Label(window, text = "")



menu_barra = Menu(window)

menu_arquivo = Menu(menu_barra, tearoff=0)
menu_arquivo.add_command(label="Help", command=help_info)

menu_barra.add_cascade(label="Menu", menu=menu_arquivo)
window.config(menu=menu_barra)


fonte_personalizada = font.Font(family="Times New Roman", size=24)
label = Label(window, text="UVB-76 Decoder", font=fonte_personalizada).place(x=130, y=100)

font_footer = font.Font(family="Times New Roman", size=12)
label = Label(window, text="1.1-rc1", font=font_footer).place(x=410, y=340)

font_decode = font.Font(family="Times New Roman", size=16)
botao_decode = Button(window,
                text="Decode",
                font=font_decode,
                command=decode_message,
                width=45,).place(x=0, y=300)


if __name__ == "__main__":
    check_new_version("1.0.0")
    window.mainloop()
